design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,493 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the storage migration between PVCs.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. Currently, the VM must be turned off and recreated, causing service disruptions and downtime.
+
+
+## Goals
+Design and develop an API to handle live storage migration between PVCs.
+
+ ## Non  Goals
+*  This feature is not directly aimed at live migrating the VM using RWO volumes. Hence, the API covers and refers only to the volume migration.
+*  This design doesn't cover migration between storage classes, but rather implements basic APIs that can be used by overlying tooling.
+*  This design doesn't include the creation of the destination PVCs, it expects the destination PVCs to exist.
+
+ ## Definition of Users
+This feature is for all KubeVirt users who want to move the PVCs assigned to a VM without workload interruptions.
+
+## User Stories
+
+  1. As a VM owner, the storage class of volumes used by a running VM is deprecated, and the volumes need to be moved to another storage class.
+  2. As a VM owner, I want to use higher performance storage class that is now available in the cluster.
+  3. As a VM owner, I see my non-resizable storage device filling up and I want to move my VM to a larger storage device without interruptions in order to give it more storage.
+  4. As an infra admin, I need to replace the storage array due to a hardware refresh and I want to do this without stopping the VMs in order to make this infra operation transparent to VM owners.
+
+## Repos
+
+  - [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+This proposal introduces a new CRD called `VolumeMigration`
+
+## Volume migration custom resource
+
+```golang
+type VolumeMigration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              VolumeMigrationSpec `json:"spec" valid:"required"`
+	Status VolumeMigrationStatus `json:"status,omitempty"`
+}
+
+type VolumeMigrationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items []VolumeMigration `json:"items"`
+}
+
+type SourceReclaimPolicy string
+
+const (
+	SourceReclaimPolicyDelete SourceReclaimPolicy = "Delete"
+	SourceReclaimPolicyRetain SourceReclaimPolicy = "Retain"
+)
+
+type MigratedVolume struct {
```

Comment:
```suggestion
type DestinationVolume struct {
```
What about this? I'm not a native speaker but using past tense here seems a bit weird.

Reply:
Not sure, I see your point but DestinationVolume looks incomplete to me as it includes more information. I would try to find a better name